website/blog/2019-09-07-managing-multiples-projects-with-lerna-and-yarn-workspaces.md

```diff
@@ -0,0 +1,181 @@
+---
+title: Managing multiples projects with Lerna and Yarn Workspaces
+author: Sergio Herrera
+authorURL: https://github.com/sergiohgz
+authorImageURL: https://avatars3.githubusercontent.com/u/14012309
+authorTwitter: sergiohgz
+---
+
+Verdaccio is a project with a big ecosystem, composed by several projects, each one with its own configurations and ways to manage. This fact transforms a simple maintenance operation, like updating a common dependency, into a real hard work.
+
+We saw a problem, these configurations entropry made harder to work with all the projects. So, we needed to simplify and unify them to make it easier. We need a **monorepo**.
+
+A monorepo is a project that can manage a collection of subprojects in a simple and unified way. There are many monorepos that Javascript developers are using daily, like [Babel](https://babeljs.io/), [Create React App](https://create-react-app.dev/) or [Material UI](https://material-ui.com/).
+
+Now, we are proud to announce our [monorepo](https://github.com/verdaccio/monorepo), our big ecosystem joined in only one repository. This is the first part of a series of articles where we will try to explain our motivation, what you need to setup your own, how to improve their management, how to work with and how to improve some workflows (CI, code quality, etc).
+
+> If you 😍 Verdaccio as we do, helps us to grow more donating to the project via [OpenCollective](https://opencollective.com/verdaccio).
+
+Thanks for support Verdaccio ! 👏👏👏👏.
+
+<!--truncate-->
+
+<div id="codefund">''</div>
+
+## Background and tools
+
+A year ago, [Juan](https://twitter.com/jotadeveloper) and I met in Madrid, Spain, and were talking about the roadmap for Verdaccio 4, the scope of the projects and more. One thing we mention is the hard work we had dealing with Verdaccio ecosystem, several projects and repositories, each one with their own dependencies, scripts, configurations, etc.
+
+We proposed the unification of the projects configuration, but that would not be realistic because we would have things replicated in all projects. We wanted to join all projects to remove a big part of them, and we found [Lerna](https://lerna.js.org/) and [Yarn Workspaces](https://yarnpkg.com/lang/en/docs/workspaces/).
+
+We discussed about having two monorepos, one for core things and tools, and other for plugins. We started both with a basic architecture and thought about the first projects that should be in the core monorepo, an ESLint config and a Babel preset. But we had things with higher priorities for Verdaccio 4, so we left them stuck.
```

Comment:
> We discussed about having two monorepos, one for core things and tools, and other for plugins.

I think this part might be excluded, since we never reached so far anyway.

>  We started both with a basic architecture and thought about the first projects that should be in the core monorepo, an ESLint config and a Babel preset. But we had things with higher priorities for Verdaccio 4, so we left them stuck.

As a first step, a basic architecture, we moved the first dependencies in the core monorepo, ESLint config and a Babel preset. 

> But we had things with higher priorities for Verdaccio 4, so we left them stuck.

:-) We might exclude this.